### PR TITLE
Simplify variable substitution tests ⚗️

### PR DESCRIPTION
**Pull request by bobcatfish**
# Changes

Each test case for parameter substitution application was being given a
totally separate test case, with the variables being used being declared
in different places across the file. For #1639 I came along and wanted
to start adding more tests for workspace substitution and found it hard
to tell where to start so I:

* Combined most of the test cases for param subsitution into one test
  so you can easily see everything that is being tested (none of the
  test cases conflicted with each other and can easily be applied
  together)
* I kept the array param test cases separate cuz they seemd to be
  testing distinct test cases
* The Volume test cases were a bit odd b/c they were trying to make sure
  substitution was _applied_ to volumes, but there is no volume specific
  function so they were calling an internal function and passing in
  dummy values that are not representative of the actual values you'd
  substitute for volumes so instead I folded these test cases into the
  param application test.

Probably the resource application test case should be made quite similar
to the param test but it seemed like some of the resource stuff was
distinct and had to be tested in isolateion (e.g. just outputs, just
inputs, etc.)

Also removed some depreated (and duplicated!) volume tests:
in #1311 I removed support for ${} but instead of removing these tests
I just updated them, making them duplicates of the above test cases.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [n/a] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [n/a] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

### CODE REVIEW

File: pkg/reconciler/taskrun/resources/apply_test.go [hunk 1]
```diff
@@ -120,38 +130,29 @@ var (
...
-			}},
+		Volumes: []corev1.Volume{{
+			Name: "$(inputs.params.FOO)",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: "$(inputs.params.FOO)",
+					},
+				},
+			}}, {
```
Comment (ImJasonH) [inline pkg/reconciler/taskrun/resources/apply_test.go]: nit: indentation might be off here, I'd expect a `}, {` at the same indentation as `Volumes:` above.

File: pkg/reconciler/taskrun/resources/apply_test.go [hunk 2]
```diff
@@ -120,38 +130,29 @@ var (
...
+					SecretName: "$(inputs.params.FOO)",
+				},
+			}}, {
+			Name: "some-pvc",
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: "$(inputs.params.FOO)",
+				},
+			},
+		},
```
Comment (ImJasonH) [inline pkg/reconciler/taskrun/resources/apply_test.go]: same nit here: equally-indented `}`s just look wrong.

File: pkg/reconciler/taskrun/resources/apply_test.go [hunk 3]
```diff
@@ -577,6 +452,61 @@ func TestApplyParameters(t *testing.T) {
...
+	}
+	dp := []v1alpha1.ParamSpec{{
+		Name:    "something",
+		Default: builder.ArrayOrString("mydefault"),
+	}, {
+		Name:    "somethingelse",
+		Default: builder.ArrayOrString(""),
+	}}
+	want := applyMutation(simpleTaskSpec, func(spec *v1alpha1.TaskSpec) {
```
Comment (ImJasonH) [inline pkg/reconciler/taskrun/resources/apply_test.go]: `applyMutation` is grooooossss and we should get rid of it. But that's not for this PR.

Comment (bobcatfish) [reply pkg/reconciler/taskrun/resources/apply_test.go]: haha its very "clever"

**Comment by bobcatfish** [comment]
I think I fixed the indentation, PTAL @vdemeester @ImJasonH  !

_Use Golang they said... No one will ever have to argue about formatting again they said..._

**Comment by vdemeester** [comment]
> _Use Golang they said... No one will ever have to argue about formatting again they said..._

Use any editor that can do `goimports+gofmt` on save :stuck_out_tongue: and it is still true, no one argues, the CI fails that's all :joy_cat: :stuck_out_tongue_closed_eyes: :angel:

**Pull request merged**